### PR TITLE
[python3] Remove usages of `six` in wptrunner and wptserve

### PR DESCRIPTION
**Pull request by stephenmcgruer**
This PR removes six usages from `tools/wptrunner` and `tools/wptserve`.

In each I removed all uses of six except for the `ensure_*` calls, which
are not trivially replaceable.

### CODE REVIEW

**Comment by stephenmcgruer** [comment]
Ping for review?

**Review by LukeZielinski — APPROVED** [review]
Sorry completely missed this.

I did go through it all, but lots of files that kind of blurred together. Trusting the checks, but maybe submit next week when people are available to handle anything that was missed? :)

**Comment by jgraham** [comment]
Oh sorry I got half way through this and then forgot to come back to it.

File: tools/wptserve/tests/functional/test_response.py [hunk 1]
```diff
@@ -177,12 +178,14 @@ def handler(request, response):
         route = ("GET", "/test/test_write_raw_content", handler)
         self.server.router.register(*route)
 
+        # TODO: I think this HAS to throw now. So it should expect
+        # BadStatusLine to be raised.
```
Comment (jgraham) [inline tools/wptserve/tests/functional/test_response.py:182]:
```
with pytest.raises(BadStatusLine) as e:
    self.request(route[1])
assert str(e) == resp_content.decode('utf-8')
```

(isn't Python scoping lovely)

File: tools/wptserve/wptserve/server.py [hunk 2]
```diff
@@ -190,8 +187,8 @@ def __init__(self, server_address, request_handler_cls,
         else:
             hostname_port = ("",server_address[1])
 
-        #super doesn't work here because BaseHTTPServer.HTTPServer is old-style
-        BaseHTTPServer.HTTPServer.__init__(self, hostname_port, request_handler_cls, **kwargs)
+        #super doesn't work here because http.server.HTTPServer is old-style
```
Comment (jgraham) [inline tools/wptserve/wptserve/server.py:190]: Old-style classes don't exist any more, so this comment doesn't make sense (I think it's fine to leave the code as-is fwiw).

**Review by jgraham — DISMISSED** [review]
Thanks for doing this! Just a few small issues.

File: tools/wptrunner/wptrunner/metadata.py [hunk 3]
```diff
@@ -443,7 +442,7 @@ def suite_start(self, data):
         self.run_info = run_info_intern.store(RunInfo(data["run_info"]))
 
     def test_start(self, data):
-        test_id = intern(ensure_str(data["test"]))
+        test_id = sys.intern(ensure_str(data["test"]))
```
[the remark this answers is not in the document]
Comment (stephenmcgruer) [reply tools/wptrunner/wptrunner/metadata.py:445]: Done.

File: tools/wptserve/tests/functional/test_response.py (hunk 1 above)
Comment (stephenmcgruer) [reply tools/wptserve/tests/functional/test_response.py:182]: Thanks!

File: tools/wptrunner/wptrunner/metadata.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (stephenmcgruer) [reply tools/wptrunner/wptrunner/metadata.py:445]: Done.

File: tools/wptserve/wptserve/server.py (hunk 2 above)
Comment (stephenmcgruer) [reply tools/wptserve/wptserve/server.py:190]: Removed comment.

**Review by jgraham — APPROVED** [review]

**Pull request merged**